requester/report.go

```diff
@@ -23,60 +23,53 @@ func (rep Report) String() string {
 	return string(b)
 }
 
-// collect pulls the records from Requester.Records as soon as
-// they are available and consumes them to build the report.
-// Returns the report when all the records have been collected.
-// Requester.collect will blocks until Requester.Records is empty.
-func (r *Requester) collect() Report {
-	rep := Report{
+// report generates and returns a Report from a previous Run.
+func (r *Requester) report() Report {
+	return Report{
```

Comment:
Should not happen but should we noop if `Run` was not called, ie `len(r.records) == 0 && r.numErr == 0` ?